core/src/com/unciv/ui/screens/cityscreen/CityScreen.kt

```diff
@@ -60,8 +61,19 @@ class CityScreen(
         const val wltkIconSize = 40f
     }
 
+    private val selectedCiv: Civilization = GUI.getWorldScreen().selectedCiv
```

Comment:
I don't like this, CityScreen should get as a parameter the viewing civ, just like other screens do

Reply:
The passing in a parameter alternative would require us to pass through an extra selectedCiv parameter to WorldTileGroup, WorldMapHolder, TileGroup, TileLayer, TileLayerCityButton, CityButton just so that it can be passed to city screen.

Are the comments at WorldScreen line 392-397 still relevant? Weird how sometimes it passes selectedCiv and other times viewingCiv.

Reply:
Not sure, that's a comment from 5 years ago so who knows?